modules/aggregation-layers/src/hexagon-layer/hexagon-aggregator.js

```diff
@@ -36,17 +36,26 @@ export function pointToHexbin({data, radius, getPosition}, viewport) {
 
   // add world space coordinates to points
   const screenPoints = [];
+  let position;
   const {iterable, objectInfo} = createIterable(data);
   for (const object of iterable) {
     objectInfo.index++;
-    screenPoints.push(
-      Object.assign(
-        {
-          screenCoord: viewport.projectFlat(getPosition(object, objectInfo))
-        },
-        object
-      )
-    );
+    position = getPosition(object, objectInfo);
+    const arrayIsFinite = Number.isFinite(position[0]) && Number.isFinite(position[1]);
+    if (arrayIsFinite) {
+      screenPoints.push(
+        Object.assign(
+          {
+            screenCoord: viewport.projectFlat(position)
+          },
+          object
```

Comment:
The original object is not needed for binning. Just push screenCoord without creating new objects.

Reply:
Addressed in latest change